src/sentry/snuba/sessions_v2.py

```diff
@@ -619,14 +619,18 @@ def isoformat_z(date):
     return datetime.utcfromtimestamp(int(to_timestamp(date))).isoformat() + "Z"
 
 
-def get_timestamps(query):
+def get_timestamps(query, rollup=None, start=None, end=None):
     """
     Generates a list of timestamps according to `query`.
     The timestamps are returned as ISO strings for now.
     """
-    rollup = query.rollup
-    start = int(to_timestamp(query.start))
-    end = int(to_timestamp(query.end))
+    if query:
+        rollup = query.rollup
+        start = query.start
+        end = query.end
+
+    start = int(to_timestamp(start))
+    end = int(to_timestamp(end))
```

Comment:
Doesn't this now always clobber the start/end values from inside the query object, even if it is None?

What's the use case for passing two different ways of specifying rollup, start and end into here anyway?

Reply:
Also, isn't sessions supposed to be transitioned to metrics? Is this part of the code even supposed to stay long term?

Reply:
i don't think it should clobber anything because the start/end values are just a reference, and we never set the start/end values inside the query object to the new values.

what i need from this function is to be able to create timestamps without using a query or `QueryDefinition`. i'm using postgres as opposed to snuba for spend viz, but to create timestamps i also need to use this function because we will be showing a graph similar to what's in orgstats v2. i'm using `get_constrained_date_range` to get `rollup`, `start`, and `end` which i then pass into this function without a query.

i thought it would be better to refactor this existing function rather than copy paste what it's doing into a new one to account for my particular case.